src/modules/accounting/accounting.py

```diff
@@ -347,13 +374,108 @@ def _calculate_rebase_report(self, blockstamp: ReferenceBlockStamp) -> RebaseRep
-        return is_bunker, finalization_share_rate, finalization_batches
+        finalization_batches, finalization_share_rate = self._get_finalization_data(blockstamp)
+        return is_bunker, finalization_batches, finalization_share_rate
+
+    def _handle_vaults_report(self, blockstamp: ReferenceBlockStamp) -> VaultsReport:
+        """
+        Generates and publishes a Merkle tree report for staking vaults at a given blockstamp.
+
+        This function performs the following steps:
+        1. Fetches staking vaults at the given block number.
+        2. If no vaults are found, returns empty report data.
+        3. Retrieves validator data and pending deposits at the refBlock.
+        4. Loads chain configuration for the target block.
+        5. Computes total values for all vaults, including validator balances and pending deposits.
+        6. Calculates vault-specific fees and slashing reserves.
+        7. Constructs structured tree data combining vault values, fees, and slashing reserves.
+        8. Builds a Merkle tree from this data.
+        9. Publishes the Merkle tree to IPFS (or similar) and retrieves a content ID (CID).
+        10. Returns the Merkle tree root and the CID as the vaults report.
+
+        Args:
+            blockstamp (ReferenceBlockStamp): Block context used to fetch vault and validator data.
+
+        Returns:
+            VaultsReport: A tuple containing the Merkle tree root (as `bytes`) and the CID (as `str`)
+                          identifying the published tree data.
+        """
+
+        vaults = self.staking_vaults.get_vaults(blockstamp.block_hash)
+        if len(vaults) == 0:
+            return ZERO_HASH, ''
+
+        current_frame = self.get_frame_number_by_slot(blockstamp)
+        validators = self.w3.cc.get_validators(blockstamp)
+        pending_deposits = self.w3.cc.get_pending_deposits(blockstamp)
+        chain_config = self.get_chain_config(blockstamp)
+        frame_config = self.get_frame_config(blockstamp)
+        simulation = self.simulate_full_rebase(blockstamp)
+
+        vaults_total_values = self.staking_vaults.get_vaults_total_values(
+            vaults=vaults,
+            validators=validators,
+            pending_deposits=pending_deposits,
+            block_identifier=blockstamp.block_hash
+        )
+
+        core_apr_ratio = calculate_gross_core_apr(
+            pre_total_ether=simulation.pre_total_pooled_ether,
+            pre_total_shares=simulation.pre_total_shares,
+            post_internal_ether=simulation.post_internal_ether,
+            post_internal_shares=simulation.post_internal_shares,
+            shares_minted_as_fees=simulation.shares_to_mint_as_fees,
+            time_elapsed_seconds=self._get_time_elapsed_seconds_from_prev_report(blockstamp),
+        )
+
+        latest_onchain_ipfs_report_data = self.staking_vaults.get_latest_onchain_ipfs_report_data(blockstamp.block_hash)
```

Comment:
maybe call `self.w3.lido_contracts.lazy_oracle.get_latest_report_data(blockstamp.block_hash)` directly here. Function with one call is redundant